paymentsheet/src/main/java/com/stripe/android/paymentsheet/state/PaymentSheetLoader.kt

```diff
@@ -196,19 +184,59 @@ internal class DefaultPaymentSheetLoader @Inject constructor(
             )
         }
 
-        val sortedPaymentMethods = async {
-            paymentMethods.await().withLastUsedPaymentMethodFirst(savedSelection.await())
+        val customer = async {
+            when (customerConfig?.accessType) {
+                is PaymentSheet.CustomerAccessType.CustomerSession -> {
+                    elementsSession.customer?.let { customer ->
+                        CustomerState(
+                            id = customer.session.customerId,
+                            ephemeralKeySecret = customer.session.apiKey,
+                            paymentMethods = customer.paymentMethods
+                                .withLastUsedPaymentMethodFirst(savedSelection.await()),
+                        )
+                    } ?: run {
+                        val exception = IllegalStateException(
+                            "Excepted 'customer' attribute as part of 'elements_session' response!"
+                        )
+
+                        errorReporter.report(
+                            ErrorReporter
+                                .UnexpectedErrorEvent
+                                .PAYMENT_SHEET_LOADER_ELEMENTS_SESSION_CUSTOMER_NOT_FOUND,
+                            StripeException.create(exception)
+                        )
+
+                        if (!stripeIntent.isLiveMode) {
```

Comment:
It might simplify this code, if you do the error reporting and validation in the `savedSelection` async. Similar to what we do for the old integration type.